x-pack/plugin/esql/qa/testFixtures/src/main/java/org/elasticsearch/xpack/esql/CsvAssert.java

```diff
@@ -337,6 +346,42 @@ private static Object rebuildExpected(Object expectedValue, Class<?> clazz, Func
         }
     }
 
+    /**
+     * Creates an equality checker that handles both single elements and lists of elements.
+     */
+    @SuppressWarnings("unchecked")
+    private static <T> BiFunction<Object, Object, Boolean> equalityChecker(
```

Comment:
Extracted this to a generic method, as we may need it for other types, like GEO types (E.g. centroid precision error?)